src/som/primitives/BlockPrims.java

```diff
@@ -52,11 +52,27 @@ public static final IndirectCallNode create() {
     return Truffle.getRuntime().createIndirectCallNode();
   }
 
+  protected interface ValuePrim {
+    void signalException(String message);
+  }
+
   @GenerateNodeFactory
   @ImportStatic(BlockPrims.class)
   @Primitive(primitive = "blockValue:", selector = "value", inParser = false,
       receiverType = {SBlock.class, Boolean.class})
-  public abstract static class ValueNonePrim extends UnaryExpressionNode {
+  public abstract static class ValueNonePrim extends UnaryExpressionNode implements ValuePrim {
+
+    @Child ExceptionSignalingNode thrower;
+
+    @Override
+    public void signalException(final String message) {
```

Comment:
Hm, can this be done as a default method to avoid code duplication?

Reply:
We can't.
We are not able to use a common superclass.
If we move the code out, methods such as insert are not visible anymore.